run_project_tests.py

```diff
@@ -1052,6 +1054,14 @@ def should_skip_rust(backend: Backend) -> bool:
             return True
     return False
 
+def should_skip_wasi(backend: Backend) -> bool:
+    if backend is not Backend.ninja:
+        return True
+    if os.environ.get('WASI') != '1':
+        return True
```

Comment:
Seems like you could remove this?

Reply:
Well, I'm not sure how to proceed here. The idea is to avoid running these tests when the LLVM installation doesn't ship a proper WASI libc, which is the case for out-of-the-box Clang installations. My original idea was to test the LLVM installation instead, but I couldn't get it done properly, so I ended up adding this. I'd be really grateful if you can lend me a hand here.

Reply:
How does one know if the LLVM installation ships a proper WASI libc? Can you query such information with llvm-config for example?

Reply:
I guess a pragmatic approach would be to perform a sanity check: look for Clang and then have it compile a single file with a main function using the proper flags to target wasm32-wasi. I remember I tried to do something similar to https://github.com/mesonbuild/meson/blob/master/run_project_tests.py#L965, but for some reason I failed. I'll try to do it again, maybe I just didn't spend too much time on it.